plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaTextDocumentServiceWraper.java

```diff
@@ -36,4 +63,45 @@ public JavaTextDocumentServiceWraper(TextDocumentService wrapped) {
           return commands;
         });
   }
+
+  public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params)
+      throws LanguageServerException {
+
+    String fileUri = params.getTextDocument().getUri();
+    FormattingOptions options = params.getOptions();
+    if (options == null) {
+      options = new FormattingOptions();
+      params.setOptions(options);
+    }
+
+    String projectPath = WsPathUtils.absolutize(extractProjectPath(fileUri));
+    String formatterPathSuffix = CHE_FOLDER + SEPARATOR + CHE_FORMATTER_XML;
+    Path projectFormatterPath = Paths.get(projectPath, formatterPathSuffix);
+    Path wsFormatterPath = Paths.get(PROJECTS_ROOT, formatterPathSuffix);
+    if (Files.exists(projectFormatterPath)) {
+      updateFormatterOptions(params, getFormatSettingsFromFile(projectFormatterPath.toFile()));
+    } else if (Files.exists(wsFormatterPath)) {
+      updateFormatterOptions(params, getFormatSettingsFromFile(wsFormatterPath.toFile()));
+    }
+    return wrapped.formatting(params);
+  }
+
+  private void updateFormatterOptions(
+      DocumentFormattingParams params, Map<String, String> options) {
+    for (String key : options.keySet()) {
+      params.getOptions().putString(key, options.get(key));
```

Comment:
params.getOptions().putAll() ?

Reply:
Can't use `putAll()` here cause the parameter must be `Map<? extends String, ? extends Either3<String, Number, Boolean>> m` but I have `Map<String,String>`